extension/public/reddit.js

```diff
@@ -5,95 +5,117 @@ var Reddit = {
   // CSS selector for the Reddit posts list
   listSelector: ".rpBJOHq2PR60pnwJlUyP0",
 
-  // HTML for hide button
-  hideLinkHTML: function () {
-    return '<div class="reddit-actions"><div class="reddit-action hide">Hide</div></div>';
-  },
+  //counter for checkForUpdate
+  counter: 0,
+  refreshFrequency: 100,
 
-  // HTML for dashboard
-  dashboardHTML: function () {
-    var h = '<div class="reddit-dashboard">';
-    h += '<div class="reddit-page-status">';
-    h +=
-      '<div class="reddit-status total_hidden"><span class="variable"></span>Total hidden</div>';
-    h += "</div>";
-    h += '<div class="reddit-controls">';
-    h += '<div class="reddit-control hide_links">Load Hide Buttons</div>';
-    h += '<div class="reddit-control unhide_links">Unhide all</div>';
-    h += "</div>";
-    h += "</div>";
-
-    return h;
+  // initalize
+  initialize: function () {
+      this.blurImage();
+      this.checkForUpdate();
   },
 
-  // initalize UI
-  initialize: function () {
-    this.setupUI();
+  //listener to scroll events
+  //updates new visible posts to be filtered
+  checkForUpdate: function(){
+      document.addEventListener('scroll', (ev) => {
+      this.counter++;
+      if(this.counter == this.refreshFrequency){
+          this.counter = 0;
+          this.blurImage();
+          console.log("updated");
+      }
+      });
   },
 
-  // initalize all UI elements and onclick binding
-  setupUI: function () {
-    list = document.querySelector(this.listSelector);
-    list.innerHTML = this.dashboardHTML() + list.innerHTML;
 
-    document
-      .querySelector(".reddit-control.unhide_links")
-      .addEventListener("click", () => {
-        this.restoreDOM();
-        console.log("unhide all");
-      });
 
-    document
-      .querySelector(".reddit-control.hide_links")
-      .addEventListener("click", () => {
-        this.setupPostUI();
-        console.log("all links hidden");
-      });
+  //check if any configured keywords are found in a post
+  //return true if found
+  async filterText (post,callBack) {
+      //temporary filter data
+      //these data must be stored in cookie
+      const blockedWordsData = [
+          "denr",
+          "nobody",
+          "taranaki",
+          "covid",
+          "isis",
+          "pizza"
+      ];
```

Comment:
Where are we sourcing this data from, it shouldn't be hard coded?

Reply:
It shouldn't be, Currently working on it to log it using cookies

Reply:
Yeah, would be best to pull it from the UI of the app into local session storage, and then read from that imo.